source/common/common/logger.cc

```diff
@@ -276,6 +276,14 @@ absl::Status Registry::setJsonLogFormat(const Protobuf::Message& log_format_stru
     return absl::InvalidArgumentError("Usage of %_ is unavailable for JSON log formats");
   }
 
+  format_as_json.replace(format_as_json.rfind("}"), 1, "%*}");
+
+  size_t flag_index = format_as_json.find("%j");
+  while (flag_index != std::string::npos) {
+    format_as_json.replace(flag_index, 2, "%+");
+    flag_index = format_as_json.find("%j");
+  }
```

Comment:
Can you add comments? I'm not sure what this does.

Reply:
done